File: LV1/1.5/eulersnumber/eulersnumber.py
def eval_euler(n):
    if n == 0:
        return 1
    if n == 1:
        return 2

    fac = [0]*(n+1)
    fac[0] = fac[1] = 1

    euler = 2
    for i in range(2, n+1):
        fac[i] = fac[i-1]*i
        euler += 1/fac[i]

    return euler


n = int(input())
euler = eval_euler(n)

print(f'{euler:.20f}')

File: LV1/1.5/eulersnumber/eulersnumber.go
package main

import "fmt"

func evalEuler(n int) float64 {
	if n == 0 {
		return 1
	}
	if n == 1 {
		return 2
	}

	fac := 1.0
	euler := 2.0

	for i := 2; i <= n; i++ {
		fac /= float64(i)
		euler += fac
	}

	return euler
}

func main() {
	var n int
	fmt.Scanf("%d", &n)
	euler := evalEuler(n)

	fmt.Printf("%.20f\n", euler+1E-16)
}
